homeassistant/components/tado/water_heater.py

```diff
@@ -0,0 +1,303 @@
+        tado, name, zone_id, supports_temperature_control, min_temp, max_temp
+    )
+
+    return entity
+
+
+class TadoWaterHeater(WaterHeaterDevice):
+    """Representation of a Tado water heater."""
+
+    def __init__(
+        self,
+        tado,
+        zone_name,
+        zone_id,
+        supports_temperature_control,
+        min_temp,
+        max_temp,
+    ):
+        """Initialize of Tado water heater entity."""
+        self._tado = tado
+
+        self.zone_name = zone_name
+        self.zone_id = zone_id
+        self._unique_id = f"{zone_id} {tado.device_id}"
+
+        self._device_is_active = False
+        self._is_away = False
+
+        self._supports_temperature_control = supports_temperature_control
+        self._min_temperature = min_temp
+        self._max_temperature = max_temp
+
+        self._target_temp = None
+
+        self._supported_features = SUPPORT_FLAGS_HEATER
+        if self._supports_temperature_control:
+            self._supported_features |= SUPPORT_TARGET_TEMPERATURE
+
+        if tado.fallback:
+            _LOGGER.debug("Default overlay is set to TADO MODE")
+            # Fallback to Smart Schedule at next Schedule switch
+            self._default_overlay = CONST_OVERLAY_TADO_MODE
+        else:
+            _LOGGER.debug("Default overlay is set to MANUAL MODE")
+            # Don't fallback to Smart Schedule, but keep in manual mode
+            self._default_overlay = CONST_OVERLAY_MANUAL
+
+        self._current_operation = CONST_MODE_SMART_SCHEDULE
+        self._overlay_mode = CONST_MODE_SMART_SCHEDULE
+
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+
+        @callback
+        def async_update_callback():
+            """Schedule an entity update."""
+            self.async_schedule_update_ha_state(True)
+
+        async_dispatcher_connect(
+            self.hass,
+            SIGNAL_TADO_UPDATE_RECEIVED.format("zone", self.zone_id),
+            async_update_callback,
+        )
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self.zone_name
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def current_operation(self):
+        """Return current readable operation mode."""
+        return WATER_HEATER_MAP_TADO.get(self._current_operation)
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temp
+
+    @property
+    def is_away_mode_on(self):
+        """Return true if away mode is on."""
+        return self._is_away
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes (readable)."""
+        return [MODE_AUTO, MODE_HEAT, MODE_OFF]
```

Comment:
We can break out this constant list to a module level constant which we return here.

Reply:
Done.